common-http/src/main/java/io/cdap/common/http/HttpResponse.java

```diff
@@ -80,6 +124,25 @@ public String getResponseBodyAsString(Charset charset) {
     return builder.build();
   }
 
+  private byte[] getResponseBodyFromStream() {
+    try {
+      if (inputStream == null) {
+        return new byte[0];
+      }
+      return ByteStreams.toByteArray(inputStream);
+    } catch (IOException e) {
+      LOG.warn("Got exception while reading input stream");
+      return new byte[0];
+    } finally {
+      Closeables.closeQuietly(inputStream);
```

Comment:
You should set the inputStream to null as well to avoid consuming again.

Reply:
done